glean-core/android/src/main/java/mozilla/telemetry/glean/scheduler/GleanLifecycleObserver.kt

```diff
@@ -39,6 +55,15 @@ internal class GleanLifecycleObserver : LifecycleEventObserver {
                 // because it belongs to the baseline ping and that ping is sent every
                 // time the app goes to background.
                 GleanBaseline.duration.start()
+
+                // Check if the "dirty bit" was set. If it was, increment the metric.
+                // Flip the dirty bit to 'true' after.
+                if (sharedPreferences.getBoolean("dirty", false)) {
```

Comment:
Shouldn't this be checking if this is true before it increments the counter?  If the flag is set to true in ON_START and then it misses the ON_STOP, it should still be true when you check it here meaning we didn't have a "clean" shutdown.

Reply:
This line is doing what you suggest: `getBoolean` will attempt to read the stored value and, if there's no stored value, will return the default value (`false`)

Reply:
Okay, I don't think I was paying attention to the fact that `false` here was the default value for if the pref doesn't exist.  Carry on :)